packages/geo-diary-v2/src/components/ui/map/gmaps.tsx

```diff
@@ -0,0 +1,375 @@
+        const isDrawDirection = destinationPoint && destinationPoint.lat && destinationPoint.lng
+        if (isDrawDirection) {
+          renderDirection({
+            destinationPoint,
+            map,
+            position,
+            googleMaps,
+            directionsService,
+            directionsRenderer,
+            travelMode,
+            onLoadedDirection,
+            currentLocation,
+            destinationAddress,
+          })
+          return
+        }
+        clearMap({ directionsRendererRef, markersRef })()
+        markers = renderMarkers({ coordinates, googleMaps, map, markerCallBack })
+        setZoomAndCenter({ googleMaps, bounds, center, zoom, map, markers: [...markers, currentLocation] })
+        markersRef.current = markers
+        return
+      },
+      () => {
+        clearMap({ directionsRendererRef, markersRef })()
+        const markers = renderMarkers({ coordinates, googleMaps, map, markerCallBack })
+        setZoomAndCenter({ googleMaps, bounds, center, zoom, map, markers })
+        markersRef.current = markers
+        return
+      },
+    )
+  }
+}
+
+export const onMarkerClick = onDrawingMarkerClick => {
+  return marker => {
+    onDrawingMarkerClick && onDrawingMarkerClick(marker)
+  }
+}
+
+export const onMarkerComplete = (onDrawingMarkerComplete, onDrawingMarkerClick) => {
+  return marker => {
+    onDrawingMarkerComplete && onDrawingMarkerComplete(marker)
+    marker.addListener('click', onMarkerClick(onDrawingMarkerClick))
+  }
+}
+
+export type OnLoadedParams = {
+  googleMaps: any
+  map: google.maps.Map
+  bounds: google.maps.LatLngBounds
+  directionsService: google.maps.DirectionsService
+  directionsRenderer: google.maps.DirectionsRenderer
+}
+
+export type HandleOnLoadedParams = {
+  googleMapsRef: React.MutableRefObject<any>
+  mapRef: React.MutableRefObject<any>
+  directionsServiceRef: React.MutableRefObject<any>
+  directionsRendererRef: React.MutableRefObject<any>
+  boundsRef: React.MutableRefObject<any>
+  onLoaded: (params: OnLoadedParams) => void
+}
+
+export const handleOnLoaded = ({
+  googleMapsRef,
+  mapRef,
+  directionsServiceRef,
+  directionsRendererRef,
+  boundsRef,
+  onLoaded,
+}: HandleOnLoadedParams) => (googleMaps, map) => {
+  googleMapsRef.current = googleMaps
+  mapRef.current = map
+  const bounds = new googleMaps.LatLngBounds()
+  const directionsService = new googleMaps.DirectionsService()
+  const directionsRenderer = new googleMaps.DirectionsRenderer()
+
+  boundsRef.current = bounds
+  directionsServiceRef.current = directionsService
+  directionsRendererRef.current = directionsRenderer
+  onLoaded && onLoaded({ googleMaps, map, bounds, directionsService, directionsRenderer })
+}
+
+export type MarkerContentProps = {
+  coordinates: CoordinateProps<any>[]
+}
+
+export const clearMap = ({ directionsRendererRef, markersRef }) => () => {
+  directionsRendererRef?.current?.setMap(null)
```

Comment:
posible error here, change to:
`directionsRendererRef?.current?.setMap && directionsRendererRef?.current?.setMap(null)`